File: factors.py
#!/usr/bin/env bash

import sys

def factorize(num):
    factors = []
    for i in range(2, num // 2 + 1):
        if num % i == 0:
            factors.append((i, num // i))
    return factors

def main():
    if len(sys.argv) != 2:
        print("Usage: factors <file>")
        return

    file_path = sys.argv[1]

    try:
        with open(file_path, 'r') as file:
            for line in file:
                num = int(line.strip())
                factor_pairs = factorize(num)
                for pair in factor_pairs:
                    print(f"{num}={pair[0]}*{pair[1]}")
    except FileNotFoundError:
        print(f"File '{file_path}' not found.")

if __name__ == '__main__':
    main()

File: factors.c
#include <stdio.h>
#include <stdlib.h>

void factorize(int number)
{
int i, p, q;
for (i = 2; i * i <= number; i++)
{
while (number % i == 0)
{
p = i;
q = number / i;
printf("%d=%d*%d\n", number, p, q);
number = q;
}
}
if (number > 1)
{
p = number;
q = 1;
printf("%d=%d*%d\n", number, p, q);
}
}

int main(int argc, char* argv[])
{
if (argc != 2) {
printf("Usage: factors <file>\n");
return (0);
}

FILE* file = fopen(argv[1], "r");
if (!file)
{
printf("Failed to open the file.\n");
return (0);
}

int number;
while (fscanf(file, "%d", &number) != EOF) {
factorize(number);
}

fclose(file);
return (0);
}
